src/OrangePaymentSlipData.php

```diff
@@ -121,12 +130,19 @@ public function getWithBankingCustomerId()
      * @param string $referenceNumber The reference number.
      * @return $this The current instance for a fluent interface.
      * @throws DisabledDataException If the data is disabled.
+     * @throws PaymentSlipException If the length exceeds 26 when adding banking customer length.
      */
     public function setReferenceNumber($referenceNumber)
     {
         if (!$this->getWithReferenceNumber()) {
             throw new DisabledDataException('reference number');
         }
+                
+        if (strlen($referenceNumber) + $this->bankingCustomerIdLength > 26) {
+            throw new PaymentSlipException('Reference number will exceed the maximum '
```

Comment:
Created separate issue for that https://github.com/ravage84/SwissPaymentSlip/issues/32